src/ship_cmd.cpp

```diff
@@ -217,19 +217,24 @@ static void CheckIfShipNeedsService(Vehicle *v)
 		default: NOT_REACHED();
 	}
 
-	const Depot *depot = FindClosestShipDepot(v, max_distance);
-
-	if (depot == nullptr) {
-		if (v->current_order.IsType(OT_GOTO_DEPOT)) {
-			v->current_order.MakeDummy();
+	if (v->current_order.IsType(OT_GOTO_DEPOT)) {
+		const Depot *depot = Depot::GetByTile(v->dest_tile);
+		if (depot) {
+			v->current_order.MakeGoToDepot(depot->index, ODTFB_SERVICE);
```

Comment:
I'm not sure this is really needed. It seems like a ship will always service at a depot if it has the chance. I'm hoping someone with a little more experience in this area reads this :)